Sming/Components/Network/src/Network/Http/Resource/HttpResourcePlugin.h

```diff
@@ -11,20 +11,64 @@
 
 #pragma once
 
-#include "HttpEventedResource.h"
 #include <Data/LinkedObjectList.h>
-#include <Network/Http/HttpServerConnection.h>
+#include "../HttpRequest.h"
+#include "../HttpResponse.h"
+#include <Data/PriorityList.h>
+
+class HttpServerConnection;
 
 class HttpResourcePlugin : public LinkedObjectTemplate<HttpResourcePlugin>
 {
 public:
 	using OwnedList = OwnedLinkedObjectListTemplate<HttpResourcePlugin>;
+	class List : public PriorityList<HttpResourcePlugin*>
+	{
+	public:
+		void add(HttpResourcePlugin* plugin)
+		{
+			if(plugin != nullptr) {
```

Comment:
A memory leak is reported here:
```valgrind
==250356==    by 0x13011A: PriorityList<HttpResourcePlugin*>::add(HttpResourcePlugin*, int) (PriorityList.h:40)
==250356==    by 0x12E389: HttpResourcePlugin::List::add(HttpResourcePlugin*) (HttpResourcePlugin.h:31)
==250356==    by 0x131910: void HttpResourcePlugin::List::add<ResourceBasicAuth*>(HttpResourcePlugin*, ResourceBasicAuth*) (HttpResourcePlugin.h:38)
==250356==    by 0x1307D1: HttpResource* HttpResourceTree::set<ResourceBasicAuth*>(String const&, Delegate<void (HttpRequest&, HttpResponse&)> const&, HttpResourcePlugin*, ResourceBasicAuth*) (HttpResourceTree.h:114)
==250356==    by 0x12F602: (anonymous namespace)::startWebServer() (application.cpp:55)
==250356==    by 0x12FA94: (anonymous namespace)::gotIP(IpAddress, IpAddress, IpAddress) (application.cpp:78)
==250356==    by 0x131AA5: std::_Function_handler<void (IpAddress, IpAddress, IpAddress), void (*)(IpAddress, IpAddress, IpAddress)>::_M_invoke(std::_Any_data const&, IpAddress&&, IpAddress&&, IpAddress&&) (std_function.h:300)
==250356==    by 0x1440AC: std::function<void (IpAddress, IpAddress, IpAddress)>::operator()(IpAddress, IpAddress, IpAddress) const (std_function.h:688)
==250356==    by 0x142B27: WifiEventsImpl::stationGotIp(IpAddress, IpAddress, IpAddress) (WifiEventsImpl.h:36)
==250356==    by 0x14349E: StationImpl::statusCallback(netif*) (StationImpl.cpp:143)
==250356==    by 0x142EA2: StationImpl::initialise(netif*)::{lambda(netif*)#1}::operator()(netif*) const::{lambda(unsigned int)#1}::operator()(unsigned int) const (StationImpl.cpp:96)
==250356== 
==250356== LEAK SUMMARY:
==250356==    definitely lost: 16 bytes in 1 blocks
```

Reply:
Not sure if this is fixed or not...

Reply:
> Not sure if this is fixed or not... steps to reproduce?

Seems to be working. But here is how you can test it yourself:
```
cd $SMING_HOME/../samples/HttpServer_Plugins
make SMING_ARCH=Host ENABLE_GDB=1 DEBUG_VERBOSE_LEVEL=3 ENABLE_MALLOC_COUNT=0
make valgrind
```

Now run in a separate terminal window:
```
curl -vvv http://username:@192.168.13.10/ip-n-auth -d "1234" -H "Content-Encoding: test"
curl -vvv http://username:password@192.168.13.10/ip-n-auth -d "1234" -H "Content-Encoding: test"
curl -vvv http://username:password@192.168.13.10/test -d "1234" -H "Content-Encoding: test"
```

And then stop the application. Valgrind will give you a memory leak report.